acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/transaction/account/TransferTransaction.java

```diff
@@ -64,33 +75,67 @@ public TransferTransaction(
   @Override
   public Hash execute(final NodeRequests node) {
     final String signedTransactionData = signedTransactionData();
-    try {
-      return Hash.fromHexString(
-          node.eth().ethSendRawTransaction(signedTransactionData).send().getTransactionHash());
-    } catch (final IOException e) {
-      throw new RuntimeException(e);
-    }
+    return sendRawTransaction(node, signedTransactionData);
+  }
+
+  @Override
+  public Hash execute(final NodeRequests node, final SignatureAlgorithm signatureAlgorithm) {
+    final String signedTransactionData =
+        signedTransactionDataWithSignatureAlgorithm(signatureAlgorithm);
+    return sendRawTransaction(node, signedTransactionData);
   }
 
   public Amount executionCost() {
     return Amount.wei(INTRINSIC_GAS.multiply(gasPrice));
   }
 
   public String signedTransactionData() {
-    final Optional<BigInteger> nonce = getNonce();
-
-    final RawTransaction transaction =
-        RawTransaction.createEtherTransaction(
-            nonce.orElse(nonce.orElseGet(sender::getNextNonce)),
-            gasPrice,
-            INTRINSIC_GAS,
-            recipient.getAddress(),
-            Convert.toWei(transferAmount, transferUnit).toBigIntegerExact());
+    final RawTransaction transaction = createRawTransaction();
 
-    return toHexString(
+    return Numeric.toHexString(
         TransactionEncoder.signMessage(transaction, sender.web3jCredentialsOrThrow()));
   }
 
+  private String signedTransactionDataWithSignatureAlgorithm(
```

Comment:
It looks like this method could be used to sign any transaction and not only a transfer transaction, is that right?
If that is the case, it would be better to move it to a utility class of some sort. Thoughts?

Reply:
You are right, I've moves it to a utility class